### ISSUE TITLE
Putting the caret on the `end given` end marker highlights more text than necessary

### ISSUE THREAD

**Issue by nikololiahim** [scalameta/metals#3901]
**Describe the bug**
Pretty much the title.

**To Reproduce** Steps to reproduce the behavior:
1) Paste the following code in VScode:
```scala
trait Stuff[A]:
  def `do`: A

given Stuff[String] with
  def `do`: String = "done"
end given

```

2) Put the caret where `end given` is.

**Expected behavior**

Only `given` should be highlighted.

**Screenshots**
![image](https://user-images.githubusercontent.com/45356788/166154090-a1ec9d24-2b89-4f0e-a7cd-71731e91d02c.png)

**Installation:**

- Operating system: Ubuntu 20 in WSL2
- VSCode version: 
```
1.66.2
dfd34e8260c270da74b5c2d86d61aee4b6d56977
x64
```
- VSCode extension version: 1.16.0
- Metals version: (found in VSCode settings, under `metals.serverVersion`): 0.11.5

**Additional context**

Highlighting of end markers for `enum`s, `object`s, and `def`s seems fine.

**Search terms**
scala 3, end marker, given

**Comment by tanishiking** [comment]
Fixed in upstream, this should be fixed in next Scala3 release :)

**Comment by tanishiking** [comment]
Fixed in Scala 3.2.0-RC1 :tada:

As a result, we won't have any highlights when the cursor is on `end given` anymore, and on `"done"` it doesn't highlight `end given` part.
This is a bit different from the requested expected behavior `Only given should be highlighted.`, but I believe this is the right behavior as `end given` doesn't have a symbol (it's a reserved keyword).

![Screen Shot 2022-06-29 at 14 03 27](https://user-images.githubusercontent.com/9353584/176355687-531cbd65-f722-4154-8358-d3dd09ee83bc.png)
